### PR TITLE
[#200] Implementing Notification list and Notification On-Click

### LINKED ISSUE #200: Implementing Notification List Page & Notification On-Click.
**Is your feature request related to a problem? Please describe.**
Putting together a Notification page that includes all the notifications the user gets. Tasks include:
- Adding the time prop to the Notification card as discussed in #152
- Adding a notification type prop to the Notification card. 
- Changing NotificationsPageController to fit the structure. 

**Describe the solution you'd like**
Make a notifications page according to the Figma design and modify/refactor code to have the time the notification was received and the type of notification as a prop.
Change the way the notifications are being populated in NotificationsPageController. 

**Describe alternatives you've considered**
N/A

**Additional context**
- This issue was brought to light, approved and allocated in the frontend team meeting on the 29th of March 2022.
- Here is a mockup of the Notification Page. The first figure (Notification List) refers to the notification page and the second figure (Notification On-Click) is the page user is redirected to when the notification is clicked on.

<p align="center">
  image
</p>

### PR DESCRIPTION
**Pull request by the1e**
# Description:
Fixes #200.  Tasks completed in this PR:

- Refactored the `NotificationCardView` to include the type of notification (follow, like, comment, share). The content of the notification changes as the type of notification changes. Tried using useState but it caused an infinite render loop error and thus decided to use a simple variable.
- Added `momentum.js` package that would display the time of the notification (i.e 1hr ago, 1 week ago etc)
- Refactored the `NotificationPageController` to the changes made in `NotificationCardView`.
- Post prop is used to redirect the user to the notification content when the notification is clicked on. 

Note: The notification endpoint is not fully implemented, we extract example data from the notification API for now, when the endpoint is implemented, we can change it when the endpoint is completely implemented.  For this reason, lines 24-25 of NotificationPageController have been commented out.
But as of now, we receive the notification with the username, handle and type of notification. If a user follows another user, this notification would be shown:

<p align="center">
  image
</p>

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [x] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by the1e** [comment]
Also, moment.js has been added to JSON but it is causing one of the tests to fail for some reason? Any help will be appreciated.

File: frontend/src/components/notifications/notificationcard/NotificationCardController.jsx [hunk 1]
```diff
@@ -1,24 +1,29 @@
+import moment from 'moment'
```
Comment (scorpionknifes) [inline frontend/src/components/notifications/notificationcard/NotificationCardController.jsx:1]:
```suggestion
// eslint-disable-next-line import/no-unresolved
import moment from 'moment'
```

Comment (scorpionknifes) [inline frontend/src/components/notifications/notificationcard/NotificationCardController.jsx:2]:
```suggestion
// eslint-disable-next-line import/no-unresolved
import moment from 'moment'
```

**Review by scorpionknifes — CHANGES_REQUESTED** [review]

File: frontend/src/components/notifications/notificationcard/NotificationCardController.jsx (hunk 1 above)
Comment (scorpionknifes) [reply frontend/src/components/notifications/notificationcard/NotificationCardController.jsx:2]: hmm, seems like i have no perms to accept this suggested, change. @arathibiju can u merge the change on github by clicking the accept change button etc.

Comment (scorpionknifes) [reply frontend/src/components/notifications/notificationcard/NotificationCardController.jsx:2]: yeah it's a shitty issue with github actions, think rebasing ur branch should solve it, but i can't be bothered

File: frontend/src/components/notifications/notificationcard/NotificationCardController.jsx [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
 import NotificationCardView from './NotificationCardView'
 
 /**
  * Creates a NotificationCard Component.
- * TO DO:
- * Implement any NotificationCardController logic if needed
+ * Due to the time variable not existing currently in the backend we are passing in a default time
+ * When the endpoint is implemented the code can be change as below and the eslint warning can be renabled
+ * time={moment(notification.time).fromNow()}
```
Comment (scorpionknifes) [inline frontend/src/components/notifications/notificationcard/NotificationCardController.jsx:9]: can we add an issue about it to the issues list?

Comment (arathibiju) [reply frontend/src/components/notifications/notificationcard/NotificationCardController.jsx:9]: yep will do

**Review by ppat965 — APPROVED** [review]
Looks good to me, changes have been made tests are passing

**Review by scorpionknifes — APPROVED** [review]

**Pull request merged**